### PR TITLE
Design iteration

### LINKED ISSUE #701: Design iteration: remaining work
Now that #678 has been merged, this issue will track the remaining work.

## Room list

- [x] When the search bar is animated, can the option to go onto a new line be hidden, and this replaced with enter? The create & join rooms flow updated this behaviour to match
- [x] Can we reduce the space between 'Invites' and All chats? What's the vertical spacing of the invites section? Maybe this can be reduced as a solution
- [x] Change invites text weight to medium
- [x] For the more menu - can the padding of this menu be moved to 16px from the right of the screen? Can the width of this menu be customised?
- [x] Change the text in the more menu to font/body/lg/regular
- [x] Change the icons in the more menu to be the outline rounded versions
- [x] Ensure the spacing between the text in the room cells and the separators match the designs
- [x] Fix color for placeholder items

### Colors

- [x] Change the skeleton UI colour to color/bg/subtle-secondary
- [x] Change the notification dot colour to color/icon/accent/tertiary
- [x] Change the search and more icon colour to color/icon/secondary
- [x] Change the background colour of the more menu to color/bg/canvas/default
- [x] Change the icon colour in the more menu to be color/icon/secondary

## Timeline

- [x] State event can miss some padding:

image

- [x] Check background of composer in light mode:

image

- [ ] #723

**bmarty: spend a lot of time to try to figure out what's happening here. There is no issue for other BottomSheet... Will treat as a separate issue...**

- [x] Increase space between 'message removed' and the timestamp so they don't overlap
- [x] Reply text size needs to be updated to match the design - in the timeline and the composer
- [ ] Maintain the same border radius of the composer when the user moves onto the second line - currently the border radius changes slightly and it looks jerky

**bmarty: We do not change the corner radius, but I guess the animation make it weird, because it introduce a vertical bar on both side**

- [x] Remove the tappable area of the timestamp - the tappable area should only be present if there is an error message
- [x] The edit UI in the composer needs to be updated to match the designs 
- [x] Disable highlighting of message for now (when editing for instance)

### Colours

- [x] Change background colour of composer text field to color/bg/subtle-secondary
- [x] Change send button colour to color/icon/accent/tertiary
- [x] Bubble colours to be updated to match the designs

## Sign in flow

https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=6904%3A148203&mode=design&t=2zhuoApgd45vGOVC-1

- [x] Change launch screen to be blank color/bg/canvas/default
- [ ] #722

**bmarty will treat as a separate issue**

### PR DESCRIPTION
**Pull request by bmarty**
More design integration, also starting to cleanup the legacy design stuff. There are still more to do on the future.

Fixes #701 (see the list of changes there) and items in [this comment](https://github.com/vector-im/element-x-android/pull/724#issuecomment-1613302222). Remaining items in this issue will be handled separately, in particular, I will start working on #722.

### CODE REVIEW

**Comment by callumu** [comment]
This all looks great and it's almost there!

A few more comments:

**Room list**
- [ ] Could we change the avatar background colour from the blue gradient to color/bg/action/primary/disabled? This is a short term solution but an improvement on the current colour
- [ ] 'Invites' doesn't look like medium text weight - can you confirm what it is?

**Timeline**
- [ ] Can you make sure that the 'Message' text in the composer is color/text/disabled?
- [ ] Can you make sure that the background colour of the composer text field is color/bg/subtle-secondary?
- [ ] Can the border colour of the composer text field be removed when it is inactive?
- [ ] There is no active state on the icon in the bottom sheet menu when selected - is this for you to change or is this part of the Message Reactions epic?
- [ ] When selecting reply or edit in the bottom sheet, can the keyboard be automatically shown?
- [ ] The distance between a message reply and the bubble message should only be 8 pixels
- [ ] Message replies in the composer and timeline should show up to two lines of text

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/atomic/atoms/UnreadIndicatorAtom.kt [hunk 1]
```diff
@@ -37,12 +37,13 @@ fun UnreadIndicatorAtom(
...
     size: Dp = 12.dp,
     color: Color = ElementTheme.colors.unreadIndicator,
+    isVisible: Boolean = true,
 ) {
     Box(
         modifier = modifier
             .size(size)
             .clip(CircleShape)
-            .background(color)
+            .background(if (isVisible) color else Color.Transparent)
```
Comment (jmartinesp) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/atomic/atoms/UnreadIndicatorAtom.kt:46]: Why not use an if to decide if the indicator should be displayed at all? Do we need to have the view taking space for some layout reasons?

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline.components_null_DefaultGroup_TimelineItemEventRowWithReplyLightPreview_0_null,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline.components_null_DefaultGroup_TimelineItemEventRowWithReplyLightPreview_0_null,NEXUS_5,1.0,en].png:1]: There is a pb here, maybe just due to the recording.

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Text.kt [hunk 2]
```diff
@@ -56,13 +56,18 @@ fun Text(
     text: String,
     modifier: Modifier = Modifier,
     color: Color = Color.Unspecified,
+    // Will be removed, only style should be used
```
Comment (jmartinesp) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Text.kt:59]: Are we sure we want to remove this customization? I agree we shouldn't use them most of the time, but there might be some cases where we might need to override them.

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline.components_null_DefaultGroup_TimelineItemEventRowWithReplyLightPreview_0_null,NEXUS_5,1.0,en].png
Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline.components_null_DefaultGroup_TimelineItemEventRowWithReplyLightPreview_0_null,NEXUS_5,1.0,en].png:1]: We might need to record them again, yes.

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline_null_DefaultGroup_TimelineViewLightPreview_0_null_4,NEXUS_5,1.0,en].png
Comment (jmartinesp) [inline tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline_null_DefaultGroup_TimelineViewLightPreview_0_null_4,NEXUS_5,1.0,en].png:1]: It's kind of weird having the 2 font colors, and the one with less contrast being used for received files events.

**Review by jmartinesp — APPROVED** [review]
LGTM, thanks for the colossal amount of work done in this PR!

I have a couple of comments, but they're just some opinions and non-blockers that can be handled in another PR, if needed.

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/atomic/atoms/UnreadIndicatorAtom.kt (hunk 1 above)
Comment (bmarty) [reply libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/atomic/atoms/UnreadIndicatorAtom.kt:46]: Yes, I did not want to change the behavior here, but I agree we could just remove the indicator. Will revisit later.

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Text.kt (hunk 2 above)
Comment (bmarty) [reply libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Text.kt:59]: If this is really necessary to override, we could still write

```kotlin
style = ElementTheme.typography.fontBodyMdMedium.copy(fontSize = 25.sp)
```

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline_null_DefaultGroup_TimelineViewLightPreview_0_null_4,NEXUS_5,1.0,en].png
Comment (bmarty) [reply tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.timeline_null_DefaultGroup_TimelineViewLightPreview_0_null_4,NEXUS_5,1.0,en].png:1]: Good catch, there is probably an issue, let me check.

**Pull request merged**